File: codingTest/programmers/python/체육복.py
def solution(n,lost,reserve):
    count = n - len(lost)
    lost.sort()
    reserve.sort()

    for l in range(len(lost)):
        for r in range(len(reserve)):
            if lost[l] == reserve[r]:
                lost[l] = -1
                reserve[r] = -1

    for item in lost:
        for r in range(len(reserve)):
            if reserve[r] == item - 1 or reserve[r] == item + 1:
                reserve[r] = -1
                break
    
    return count + len(list(filter(lambda x: x == -1, reserve)))

File: codingTest/programmers/javascript/체육복.js
const solution = (n, lost, reserve) => {
  let count = n - lost.length;
  lost.sort((a, b) => a - b);
  reserve.sort((a, b) => a - b);

  for (const l in lost)
    for (const r in reserve)
      if (reserve[r] == lost[l]) (reserve[r] = -1), (lost[l] = -1);

  lost.forEach(l => {
    for (const i in reserve)
      if (reserve[i] == l - 1 || reserve[i] == l + 1) {
        reserve[i] = -1;
        break;
      }
  });
  return count + reserve.filter(e => e == -1).length;
};
